include/struct_defs/box_pokemon.h

```diff
@@ -1,22 +1,25 @@
 #ifndef POKEPLATINUM_BOX_POKEMON_H
 #define POKEPLATINUM_BOX_POKEMON_H
 
-#include "struct_defs/pokemon_substruct_1.h"
-#include "struct_defs/pokemon_substruct_2.h"
-#include "struct_defs/pokemon_substruct_3.h"
-#include "struct_defs/pokemon_substruct_0.h"
+#include "struct_defs/pokemon_data_block_b.h"
+#include "struct_defs/pokemon_data_block_c.h"
+#include "struct_defs/pokemon_data_block_d.h"
+#include "struct_defs/pokemon_data_block_a.h"
 
+/**
+ * @brief Boxed Pokemon data structure
+ */
 typedef struct {
-    u32 unk_00;
-    u16 unk_04_0 : 1;
-    u16 unk_04_1 : 1;
-    u16 unk_04_2 : 1;
+    u32 personality; //!< The personality value of the pokemon
+    u16 partyDecrypted : 1; //!< Whether the pokemons party data is currently decrypted
+    u16 boxDecrypted : 1; //!< Whether the pokemons boxed data is currently decrypted
+    u16 invalidData : 1; //!< Set if the pokemons data is invalid (failed checksum). This will make the mon appear as a Bad Egg in game
     u16 unk_04_3 : 13;
-    u16 unk_06;
-    u8 unk_08[sizeof(PokemonSubstruct0)
-              + sizeof(PokemonSubstruct1)
-              + sizeof(PokemonSubstruct2)
-              + sizeof(PokemonSubstruct3)];
+    u16 checksum; //!< Checksum for the pokemons data
+    u8 dataBlocks[sizeof(PokemonDataBlockA)
+              + sizeof(PokemonDataBlockB)
+              + sizeof(PokemonDataBlockC)
+              + sizeof(PokemonDataBlockD)]; //!< Data blocks containing the pokemons boxed data, stored encrypted
```

Comment:
These blocks are all the same size, if I remember correctly. You could clean this up a bit by doing this:

```c
typedef union {
    PokemonDataBlockA blockA;
    PokemonDataBlockB blockB;
    PokemonDataBlockC blockC;
    PokemonDataBlockD blockD;
} PokemonDataBlock;

typedef struct BoxPokemon {
    // other fields...
    PokemonDataBlock dataBlocks[4];
} BoxPokemon;
```